### ISSUE TITLE
CalculateStatistics: Add SSMD measure

### ISSUE THREAD

**Issue by braymp** [CellProfiler/CellProfiler#1574]
From SBI2, the consensus is that the SSMD is preferred to the Z'-factor. The AGM says CP calculates it, but it currently doesn't, so we should add it ASAP :smile: 

The paper describing it is here: http://www.sciencedirect.com/science/article/pii/S0888754307000079

**Comment by dlogan** [comment]
Related to #65

**Comment by LeeKamentsky** [comment]
Assigned to Allen since he said he wanted to do it.

**Comment by AnneCarpenter** [comment]
Note that Bartek Rajwa captures some of his exhaustive knowledge of assay statistics in this article, which will guide us on what statistic(s) to add:

Effect-Size Measures as Descriptors of Assay Quality in High-Content Screening: A Brief Review of Some Available Methodologies

Rajwa Bartek. ASSAY and Drug Development Technologies. October 2016, doi:10.1089/adt.2016.740

**Comment by 0x00b1** [comment]
Nice

**Comment by 0x00b1** [comment]
I talked to Bartek about this last week. It might be a good feature to add for the next release.

Is there still interest? @braymp? @AnneCarpenter? @dlogan?

**Comment by AnneCarpenter** [comment]
On the one hand, it's embarrassing to not have the recommended statistic in CP. 

On the other hand, no end user has asked us for it, I think ever.

On the other hand, it can't be that hard to calculate a statistic. 

So I've no clear opinion!

**Comment by 0x00b1** [comment]
@AnneCarpenter Good point!

I don’t think it’s that difficult from my SBI2 notes (and the Wikipedia article), e.g.

```Python
def ssmd(positive, negative):
    mean = numpy.mean(positive) - numpy.mean(negative)

    standard_deviation = numpy.sqrt(numpy.sd(positive)^2 + numpy.sd(negative)^2)

    return mean / standard_deviation
```

**Comment by 0x00b1** [comment]
A quick aside, `CalculateStatistics` is an unexpected name for this module. I would assume, from the name, it’d compute summary statistics rather than effect size measurements.

**Comment by AnneCarpenter** [comment]
Yeah, I'd say the common terms for what it measures is "assay quality" (that's Z' and V) and "dose response" (that's EC50). I agree Statistics is pretty generic but I can't think of a better term that covers experiment-level statistics like these. One could make an argument for "CalculateExperimentStatistics" but then again the EC50 is per-treatment not per-experiment. So, yeah, no good solution I can think of.

**Comment by dlogan** [comment]
I think this is a good idea to have SSMD (and possibly others as noted above).  But yeah, folks haven't clamored for it.  Mostly, it would give a sense of completeness, such that the biologist could in theory use CP end-to-end rather than exporting data to Excel to simply calculate SSMD, etc.

Maybe "CalculateAssayQuality" or "CalculateEffectSize" would be better names?

**Comment by braymp** [comment]
I like "CalculateAssayQuality"

**Comment by AnneCarpenter** [comment]
Well it kind of leaves out EC50, but if you'd all considered that and nobody minds that I won't object.

**Comment by ErinWeisbart** [comment]
CellProfiler doesn't calculate summary statistics (like Z' factor) any more, so we are closing this issue!
